enhancements/sig-architecture/123-addon-multiple-configs/README.md

```diff
@@ -0,0 +1,245 @@
+
+### Non-Goals
+
+- `ClusterManagementAddon` `spec.supportedConfigs` does not allow the user to configure multiple addon configs with the same GVK in the `defaultConfig`.
+
+## Proposal
+
+### User Stories
+
+1. The admin can configure the `ClusterManagementAddon` API to set multiple configs with the same GVK per install strategy.
+2. The admin can configure the `ManagementClusterAddon` API to set multiple configs with the same GVK for a specific cluster.
+
+## Design Details
+
+### API changes
+
+No API changes are needed, `cma.spec.installStrategy.placements[].configs` and `mca.spec.configs` already support multiple configs. 
+
+```go
+	// Configs is the configuration of managedClusterAddon during installation.
+	// User can override the configuration by updating the managedClusterAddon directly.
+	// +optional
+	Configs []AddOnConfig `json:"configs,omitempty"`
+```
+
+### Effective addon configs
+
+As the motivation mentioned, today there are 3 places to set addon configurations. 
+
+Currently, since each GVK supports only one config, the GVK is used as the identifier for configs. When configs with the same GVK configured in 
+multiple places, it follows the override sequence and only one config is listed as the effective config in the `mca.status`. 
+
+With the support for multiple configs with the same GVK, `GVK + namespace + name` becomes the identifier. 
+We need to reconsider the logic for setting `mca.status` when configuring multiple configs in multiple places.
+
+In the initial discussion we listed three options, finally choose the below "Override by GVK" option, as it satisfies more on 
+the real use case than the other two. The other two are listed in the **Alternatives** for future needs.
+
+The below example omits the group and namespace for simplicity. 
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: example
+spec:
+  supportedConfigs:
+    - resource: r1
+      defaultConfig:
+        name: n1
+    - resource: r2
+  installStrategy:
+    type: Placements
+    placements:
+      - name: example
+        namespace: example
+        configs:
+          - resource: r2
+            name: n1
+          - resource: r2
+            name: n2
+
+
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: example
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-agent-addon
+  configs:
+    - resource: r2
+      name: n3
+```
+
+#### Override by GVK
+
+"Override by GVK" is a way try to keep consistent with current behavior. GVK is used as the identifier for a group of configs.
+
+In the above example, config `r2` `n3` in `ManagedClusterAddOn` (place `3`) will override `r2` `n1` and `n2` in `installStrategy` (place `2`). 
+And for default config `r1` `n1`, since no other `r1` resource defined in place `2` and `3`, it will be kept in the effective addon configs.
+The `mca.status` will be like:
```

Comment:
```suggestion
In the above example, config `r2` `n3` in `ManagedClusterAddOn` (place `3`) will override `r2` `n1` and `n2` in `installStrategy` (place `2`). The default config `r1` `n1` will remain, since no other config for resource `r1` is defined in either place `2` or `3`.
The `mca.status` will look like the following:
```